client/jetpack-onboarding/main.jsx

```diff
@@ -35,7 +37,25 @@ class JetpackOnboardingMain extends React.PureComponent {
 		stepName: STEPS.SITE_TITLE,
 	};
 
-	// TODO: Add lifecycle methods to redirect if no siteId
+	componentDidMount() {
+		const { siteId, siteSlug } = this.props;
+
+		// If we are missing the Jetpack onboarding credentials,
+		// redirect back to wp-admin so we can obtain them again.
+		if ( ! siteId && siteSlug ) {
+			const siteDomain = siteSlug.replace( '::', '/' );
```

Comment:
Hmm, I'd rather not manually re-generate the domain from the slug, without even using a library function -- it'll be super-hard to locate this if we miss e.g. an edge case, or the implementation of the slug-to-domain mapping is ever going to allow for different cases we're not yet considering.

Can we use [this mechanism](https://github.com/Automattic/wp-calypso/blob/e373aec638e9acca56fd0c05b4a267a0a176baec/client/state/selectors/get-unconnected-site-id-by-slug.js#L21-L24) instead? That uses our existing dictionary of unconnected site URLs and a lib function instead...

That will also include the protocol and thus allow us to drop the `//` on L53, which might also cause trouble otherwise...

Reply:
> Can we use this mechanism instead? That uses our existing dictionary of unconnected site URLs and a lib function instead...

Scratch the part about the 'existing dictionary of unconnected site URLs', we don't have that here...